src/NServiceBus.Core/Recoverability/RecoverabilityPipelineExecutor.cs

```diff
@@ -15,7 +15,7 @@ class RecoverabilityPipelineExecutor<TState> : IRecoverabilityPipelineExecutor
             MessageOperations messageOperations,
             RecoverabilityConfig recoverabilityConfig,
             Func<ErrorContext, TState, RecoverabilityAction> recoverabilityPolicy,
-            Pipeline<IRecoverabilityContext> recoverabilityPipeline,
+            IPipeline<IRecoverabilityContext> recoverabilityPipeline,
```

Comment:
This change might impact perf due to the virtual dispatch. Maybe not a big deal becauss in other places we also use the interface but I wanted to mention it

Reply:
I think the only alternative to make this somewhat testable then would be to use subclassed implementations of `Pipeline<T>` for testing purposes which seems to be more effort.

I'd say this potential impact is acceptable?

Reply:
I'd agree